ts/main.ts

```diff
@@ -33,19 +34,40 @@ $photoUrlInput.addEventListener('input', () => {
 $form.addEventListener('submit', (event: Event) => {
   event.preventDefault();
 
-  const newEntry: JournalEntry = {
-    entryId: data.nextEntryId,
+  const entryData = {
     title: $titleInput.value,
     photoUrl: $photoUrlInput.value,
     notes: $notesInput.value,
   };
-
-  data.entries.unshift(newEntry);
-  data.nextEntryId++;
-
-  const $entry = renderEntry(newEntry);
-  $entriesList.prepend($entry);
-
+  if (data.editing) {
+    const editedEntry: JournalEntry = {
+      ...entryData,
+      entryId: data.editing.entryId,
+    };
+    const index = data.entries.findIndex(
+      (entry) => entry.entryId === data.editing!.entryId,
+    );
+    if (index !== -1) {
+      data.entries[index] = editedEntry;
+      const $editedEntry = renderEntry(editedEntry);
+      const $originalEntry = $entriesList.querySelector(
+        `.entry-item[data-entry-id="${data.editing!.entryId}"]`,
+      );
+      if ($originalEntry) {
+        $originalEntry.replaceWith($editedEntry);
```

Comment:
Just leaving the comment to be consistent but you should always be checking if the element is falsy so that we can throw an error and see that it went wrong. If something goes wrong here, we won't know because there's not message telling us that.

```if (!$originalEntry) throw new Error('original entry not found')```